mater/lib/src/v1/reader.rs

```diff
@@ -45,19 +46,38 @@ pub(crate) async fn read_block<R>(mut reader: R) -> Result<(Cid, Vec<u8>), Error
 where
     R: AsyncRead + Unpin,
 {
-    let full_block_length: usize = read_varint(&mut reader).await?;
-    let mut full_block_buffer = vec![0; full_block_length];
-    reader.read_exact(&mut full_block_buffer).await?;
+    let (full_block_length, _): (u64, usize) = read_varint(&mut reader).await?;
+    let (cid, cid_bytes_read) = Cid::read_bytes_async(&mut reader).await?;
 
-    // We're cheating to get Seek
-    let mut full_block_cursor = Cursor::new(full_block_buffer);
-    let cid = Cid::read_bytes(&mut full_block_cursor)?;
+    let data_size = full_block_length as usize - cid_bytes_read;
+    let mut data_buffer = vec![0; data_size];
+    reader.read_exact(&mut data_buffer).await?;
 
-    let data_start_position = full_block_cursor.position() as usize;
-    let mut full_block_buffer = full_block_cursor.into_inner();
+    Ok((cid, data_buffer))
+}
+
+pub(crate) async fn skip_block<R>(mut reader: R) -> Result<BlockMetadata, Error>
```

Comment:
I feel that this is more accurate
```suggestion
pub(crate) async fn read_block_metadata<R>(mut reader: R) -> Result<BlockMetadata, Error>
```